clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java

```diff
@@ -261,7 +261,11 @@ public void setChunkingInfo(
         record.setValueManifest(chunkedValueManifest);
         record.setRmdManifest(chunkedRmdManifest);
       } else {
-        LOGGER.error("Transient record is missing when trying to update value/RMD manifest.");
+        String msg = "Transient record is missing when trying to update value/RMD manifest for topic "
+            + ingestionTask.getKafkaVersionTopic() + " partition:" + subPartition;
+        if (partitionConsumptionState.isEndOfPushReceived() && !REDUNDANT_LOGGING_FILTER.isRedundantException(msg)) {
```

Comment:
nit: It is also okay to move the EOP check all the way up, but it is okay to keep it here as long as we don't over flow the log.

Reply:
I would prefer moving the EOP check up as you mentioned... no need to allocate the string if that check short-circuits.